core/web/pipeline_runs_controller.go

```diff
@@ -84,30 +85,49 @@ func (prc *PipelineRunsController) Create(c *gin.Context) {
 	}
 	idStr := c.Param("ID")
 
+	user, isUser := authenticatedUser(c)
+	ei, _ := authenticatedEI(c)
+	authorizer := webhook.NewAuthorizer(prc.App.GetStore().DB, user, ei)
+
 	// Is it a UUID? Then process it as a webhook job
 	jobUUID, err := uuid.FromString(idStr)
 	if err == nil {
-		jobRunID, err2 := prc.App.RunWebhookJobV2(context.Background(), jobUUID, string(bodyBytes), pipeline.JSONSerializable{Null: true})
+		canRun, err2 := authorizer.CanRun(jobUUID)
 		if err2 != nil {
 			jsonAPIError(c, http.StatusInternalServerError, err2)
 			return
 		}
-		respondWithPipelineRun(jobRunID)
+		if canRun {
+			jobRunID, err3 := prc.App.RunWebhookJobV2(context.Background(), jobUUID, string(bodyBytes), pipeline.JSONSerializable{Null: true})
```

Comment:
Looks like this is a blocking call. We could either pass in the c.Request.Context() so the user can cancel OR we could make this async

Reply:
Good call